packages/Algernon.package/ALGWorldMenuType.class/class/menuEntriesNew.st

```diff
@@ -1,5 +1,14 @@
 algertem lists
 menuEntriesNew
 
-	^ (ALGNamedWorldMenuEntry newFrom: ActiveWorld mainDockingBars first)
-		subMenus
\ No newline at end of file
+	|menuEntries entries|
+	
+	entries := OrderedCollection new.
+	
+	menuEntries := (self dockingBar submorphs
+		collect: [:morph | ALGNamedWorldMenuEntry newFrom: morph])
+		select: [:entry | entry isValid].
+	entries addAll: menuEntries.
+
+	^ menuEntries
+		inject: OrderedCollection new into: [:accumulator :entry | accumulator addAll: entry subEntries; yourself]
```

Comment:
Here, the `inject:into:` pattern makes sense, but you probably meant to use `entries` somehow